vendor/github.com/rancher/types/config/context.go

```diff
@@ -180,6 +187,7 @@ type UserContext struct {
 	UnversionedClient rest.Interface
 	APIExtClient      clientset.Interface
 	K8sClient         kubernetes.Interface
+	runContext        context.Context
```

Comment:
What is the purpose of a this new Context?

Reply:
It's not new, it's the main context rancher starts with and needs to be passed into the management controller startup, if using the user context then when a cluster is deleted management controllers would also shut down (was a fun issue to find)

Reply:
To clarify, it's not a context like Ranchers (management, cluster, project) that's a go context usually seen as `ctx`

Reply:
Ahh, its the context from https://github.com/rancher/rancher/blob/2c9f04baf3afcba07bfed94cf41a606a2ebd2a71/main.go#L204